src/components/Search/SearchInputAndResults.tsx

```diff
@@ -82,12 +88,9 @@ export default function SearchInputAndResults({
       let searchParameters;
       if (!searchTerm) {
         searchParameters = { repository };
-      } else if (emailMatch.test(searchTerm)) {
+      } else if (authorInfoMatch.test(searchTerm)) {
         searchParameters = { repository, author: searchTerm };
-      } else if (
-        longHashMatch.test(searchTerm) ||
-        shortHashMatch.test(searchTerm)
-      ) {
+      } else if (minimumCharacterLength.test(searchTerm)) {
```

Comment:
as we discussed, use `searchTerm.length < 3` as a first if condition to set the error, as it's more readable, and then use the regexp as it was before.